onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -0,0 +1,446 @@
+        if input_value is None:
+            # This input argument is None, which is mapped
+            # to a NULL value in TorchScript type system.
+            torch_value = self._graph.op("prim::Constant")  # type: ignore[attr-defined]
+            torch_value.setType(torch._C.OptionalType.ofTensor())
+            return torch_value
+        torch_value = self._graph.addInput(input_name)
+        torch_value.setType(torch._C.TensorType.create_from_tensor(input_value))
+        tensor_value = _wrap_torch_value_to_tensor(torch_value)
+        return tensor_value
+
+    @beartype
+    def register_outputs(self, outputs: TorchScriptTensor | tuple[TorchScriptTensor, ...]):
+        unwrapped_outputs = _unwrap_tensors_to_torch_values(outputs)
+        if isinstance(unwrapped_outputs, torch.Value):
+            self._graph.registerOutput(unwrapped_outputs)
+            return
+        assert isinstance(unwrapped_outputs, Sequence)
+        for ts_output in unwrapped_outputs:
+            assert isinstance(
+                ts_output, torch.Value
+            ), f"ts_output must be a torch.Value, not {type(ts_output)}"
+            self._graph.registerOutput(ts_output)
+        return
+
+    def _add_constant_to_graph(self, constant) -> torch.Value:
+        if isinstance(constant, float):
+            return _create_torchscript_op(
+                self._graph,
+                "onnx::Constant",
+                inputs=(),
+                attributes=dict(value=torch.tensor(constant, dtype=torch.float)),
+            )[0]
+        if isinstance(constant, int):
+            return _create_torchscript_op(
+                self._graph,
+                "onnx::Constant",
+                inputs=(),
+                attributes=dict(value=torch.tensor(constant, dtype=torch.int64)),
+            )[0]
+        if constant is None:
+            value = _create_torchscript_op(
+                self._graph, "prim::Constant", inputs=(), attributes={}
+            )[0]
+            value.setType(torch.OptionalType.ofTensor())
+            return value
+        if isinstance(constant, (tuple, list)) and all(
+            isinstance(val, int) for val in constant
+        ):
+            return _create_torchscript_op(
+                self._graph,
+                "onnx::Constant",
+                inputs=(),
+                attributes=dict(value=torch.tensor(constant, dtype=torch.int64)),
+            )[0]
+        if isinstance(constant, (tuple, list)) and all(
+            isinstance(val, float) for val in constant
+        ):
+            return _create_torchscript_op(
+                self._graph,
+                "onnx::Constant",
+                inputs=(),
+                attributes=dict(value=torch.tensor(constant, dtype=torch.float)),
+            )[0]
+
+        raise TypeError(
+            f"Constant input `{constant}` of type '{type(constant)}' is not supported"
+        )
+
+    @beartype
+    def _add_torchscript_op(
```

Comment:
nit: `_add_torchscript_op_call` may help distinguish an op from an invocation of an op.

Reply:
Similarly for other methods below

Reply:
SG!

Reply:
Done